app/controllers/revisions_controller.rb

```diff
@@ -46,13 +46,21 @@ def trash
   end
 
   def recover
-    version = RecoverableVersion.find(params[:id])
-    class_name = object.class.name.humanize
+    version    = RecoverableVersion.find(params[:id])
+    class_name = if version.object.is_a?(Note) && version.object.node_id == Node.issue_library.id
```

Comment:
Good idea: done. But pleeeease can we fix this whole Issue/Note thing soon. It's giving me nightmares 😿